frontend/src/components/DragAndDrop/DragAndDrop.js

```diff
@@ -0,0 +1,144 @@
+import { useState, useRef, useEffect } from "react";
+import { CloudArrowUpFill } from "react-bootstrap-icons";
+import { useSpring, animated } from "react-spring";
+import "./DragAndDrop.css";
+import NavBar from "../NavBar";
+import App from "../../App";
+
+function DragAndDrop() {
+  const [dragActive, setDragActive] = useState(false);
+  const [fileUploaded, setFileUploaded] = useState(false);
+  const [invalidJson, setInvalidJson] = useState(false);
+  const [lines, setLines] = useState();
+  const inputRef = useRef(null);
+
+  const flyingAnimation = useSpring({
+    transform: fileUploaded ? "translateY(-100%)" : "translateY(0%)",
+  });
+
+  const handleDrag = function (e) {
```

Comment:
Why not split it to 2 different functions if you need to check for event type inside?

Reply:
If split, then three separate attributes will be required to be added to the form element!
BTW done!